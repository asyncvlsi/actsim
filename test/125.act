function testme (int a) : int
{
  int b[3]; 
  chp {
    b[a] := a;
    self := a
  }
}

defproc test()
{
  int x[4], y[3];

  chp {
    (;i:3: y[i] := i);
    (;i:4: x[i] := 2*i);
    x[2] := 2;
    log ("got: ", testme (x[2]))
  }
}
