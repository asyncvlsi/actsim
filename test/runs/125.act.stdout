[                  80] <>  got: 2
